packages/core/src/modules/dids/domain/DidDocument.ts

```diff
@@ -214,3 +215,38 @@ export function keyReferenceToKey(didDocument: DidDocument, keyId: string) {
 
   return key
 }
+
+/**
+ * Extracting the first publicKeyBase58 for signature type
+ * @param type Signature type
+ * @param didDocument DidDocument
+ * @returns publicKeyBase58
+ */
+export async function getPublicKeyBase58(type: string, didDocument: DidDocument): Promise<string> {
```

Comment:
Maybe we can make this a bit more generic so it's more applicable to other use cases. What if we call it `findVerificationMethodByKeyType`? It won't extract the public key base 58, but that can be done in a specific util method (`getIndyDidFromVerficationMethod`).

Reply:
Resolved